cinema/views.py

```diff
@@ -56,3 +84,54 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie_id = self.request.query_params.get("movie")
+        if movie_id:
+            queryset = queryset.filter(movie_id=movie_id)
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related()
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows")
+                        * F("cinema_hall__seats_in_row")
+                        - Count("tickets"))
```

Comment:
Ensure that the calculation of `tickets_available` correctly reflects the actual number of available tickets. This annotation assumes that all tickets are sold for each seat, which might not be the case if there are other constraints or conditions.